models/stock_warehouse_weather.py

```diff
@@ -0,0 +1,84 @@
+from odoo import fields, models, api
+from datetime import datetime
+from odoo.exceptions import ValidationError
+import requests
+
+class FlowerShopWeather(models.Model):
+    _name = 'flower.shop.weather'
+    _description = 'weather checking for plants'
+
+    warehouse_id = fields.Many2one('stock.warehouse')
+    temperature = fields.Float("Temperature")
+    pressure = fields.Float()
+    humidity = fields.Float()
+    wind_speed = fields.Float()
+    rain_volume = fields.Float()
+    description = fields.Char()
+    capture_time = fields.Datetime()
+    
+class warehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    weather_ids = fields.One2many('flower.shop.weather', 'warehouse_id')
+    
+    def _get_api_key_and_location(self, show_error):
+        api_key = self.env["ir.config_parameter"].sudo().get_param("flower_shop.weather_api_key")
+        if api_key == "unset" or not api_key:
+            show_error(True)
+        return api_key
+
+    def get_weather(self, show_error=True):
+        self.ensure_one()
+        api_key = self._get_api_key_and_location(show_error)
+        url = "https://api.openweathermap.org/data/2.5/weather?lat=25.2582&lon=55.3047&appid={}".format(api_key)
+        try:
+            response = requests.get(url, timeout=10)
+            response.raise_for_status()
+            entries = response.json()
+            self.env["flower.shop.weather"].create({
+                "warehouse_id": self.id,
+                "description": entries["weather"][0]["description"],
+                "pressure": entries["main"]["pressure"],
+                "temperature": entries["main"]["temp"],
+                "humidity": entries["main"]["humidity"] / 100,
+                "wind_speed": entries["wind"]["speed"],
+                "rain_volume": entries["rain"]["1h"] if "rain" in entries else 0,
+                "capture_time": fields.Datetime.now(),
+            })
+        except Exception as e:
+            raise ValidationError(e)
```

Comment:
```suggestion
         except Exception as e:
            msg = _("ERROR in Warehouse ID {}: {}".format(self.id, e))
            _logger.exception(msg)
            if show_error:
                raise UserError(msg)
```
Here is an example of how the `show_error` flag is used. When this method is invoked from a scheduled action, we should set the flag to `False`. Otherwise, when the scheduled action invokes this method, an error will be raised and the whole transaction will be rolled back. When multiple transactions fail on Odoo SH, the SH team disables the scheduled action to save on their resources from being utilized in this erroneous manner.

I would like to share an amazing video from Odoo Belgium's RnD team on how to write resilient scheduled actions: https://www.youtube.com/watch?v=CLdeQLxwJPU&pp=ygUNY3JvbiBqb2Igb2Rvbw%3D%3D